amundsen_application/static/js/ducks/search/api/v0.ts

```diff
@@ -1,24 +1,36 @@
-import axios, { AxiosResponse, AxiosError } from 'axios';
+import axios, { AxiosError, AxiosResponse } from 'axios';
 
-import { SearchResponse } from '../types';
+import { SearchAllRequest, SearchResponse, SearchResourceRequest } from '../types';
 
-import { SearchReducerState } from '../reducer';
-
-function transformSearchResults(data: SearchResponse): SearchReducerState {
-  return {
-    searchTerm: data.search_term,
-    dashboards: data.dashboards,
-    tables: data.tables,
-    users: data.users,
-  };
+export function searchAll(action: SearchAllRequest) {
+  const { term, options } = action;
+  let baseUrl = '/api/search/v0';
+  return axios.all([
+      axios.get(`${baseUrl}/table?query=${term}&page_index=${options.tableIndex || 0}`),
+      axios.get(`${baseUrl}/user?query=${term}&page_index=${options.userIndex || 0}`),
+    ]).then(axios.spread((tableResponse: AxiosResponse<SearchResponse>, userResponse: AxiosResponse<SearchResponse>) => {
+      return {
+        searchTerm: tableResponse.data.search_term,
+        tables: tableResponse.data.tables,
+        users: userResponse.data.users,
+      }
+  })).catch((error: AxiosError) => {
+    // TODO - handle errors
+  });
 }
 
-export function searchExecuteSearch(action) {
-  const { term, pageIndex } = action;
-  return axios.get(`/api/search/v0/table?query=${term}&page_index=${pageIndex}`)
-  .then((response: AxiosResponse<SearchResponse>) => transformSearchResults(response.data))
-  .catch((error: AxiosError) => {
-    const data = error.response ? error.response.data : {};
-    return transformSearchResults(data);
-  });
+
+export function searchResource(action: SearchResourceRequest) {
+  const { term, pageIndex, resource } = action;
+  return axios.get(`/api/search/v0/${resource}?query=${term}&page_index=${pageIndex}`)
+    .then((response: AxiosResponse) => {
+      const { data } = response;
+      return {
```

Comment:
I think it makes sense here for this method to return just the resource and the results of that resource. see comment in the `reducer.ts`